consensus/src/lib.rs

```diff
@@ -50,11 +66,113 @@ pub enum ConsensusCreationError {
     MiningStopped,
     #[error("Mining failed")]
     MiningFailed,
+    #[error("Staking error")]
+    StakingError(#[from] ConsensusPoSError),
+    #[error("Staking failed")]
+    StakingFailed,
+    #[error("Staking stopped")]
+    StakingStopped,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+pub enum GenerateBlockInputData {
+    PoW,
+    PoS(Box<PoSGenerateBlockInputData>),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+pub struct PoSGenerateBlockInputData {
+    stake_private_key: PrivateKey,
+    vrf_private_key: VRFPrivateKey,
+    pool_id: PoolId,
+    kernel_input: TxInput,
+    kernel_witness: InputWitness,
```

Comment:
Having a witness here doesn't make sense. Isn't the private key here to produce that witness?

Reply:
I've removed `kernel_witness` and have updated all of the kernel signing code.